commercetools-models/src/test/java/io/sphere/sdk/categories/CategoryDraftTest.java

```diff
@@ -21,7 +21,7 @@ public void categoryDraftImplementsWithKey() {
         assertThat(key).isEqualTo(category.getKey());
     }
 
-    private <T extends WithKey> String getKey(@Nonnull final T resource) {
+    public <T extends WithKey> String getKey(@Nonnull final T resource) {
```

Comment:
@acbeni I am wondering why would we change this `private` utils to `public`? 🤔

Reply:
it's for the way we package our tests, actually we use  a [custom plugin](https://github.com/commercetools/commercetools-jvm-sdk/tree/master/osgi-support/mirror-subclass-plugin) to subclass all the tests and this is done in order to add some annotations that are specific to [PaxExam](https://ops4j1.jira.com/wiki/spaces/PAXEXAM4/overview) via byte code manipulation, these annotations make sure that the test is run via pax exam runner as in the [example](https://ops4j1.jira.com/wiki/spaces/PAXEXAM3/pages/27295748/Getting+Started+with+OSGi+Tests), all these extensions/subclassing operations makes it difficult to work with private methods, and this is why i switch methods (in test) to public, it's not a big deal since they are not shipped to our clients, and in general it gives us one less aspect to take care of.

Reply:
hmm, thanks for this explanation @acbeni  Although, this is confusing for me given the error messages are:
```
categoryDraftImplementsWithKey(io.sphere.sdk.categories.___CategoryDraftTest)  Time elapsed: 0.04 sec  <<< ERROR!
java.lang.ClassNotFoundException: org.mockito.Mockito not found

productDraftImplementsWithKey(io.sphere.sdk.products.___ProductDraftTest)  Time elapsed: 0.026 sec  <<< ERROR!
java.lang.ClassNotFoundException: org.mockito.Mockito not found

productLikeImplementsWithKey(io.sphere.sdk.products.___ProductLikeTest)  Time elapsed: 0.014 sec  <<< ERROR!
java.lang.ClassNotFoundException: org.mockito.Mockito not found 
```

Reply:
yep figured out i need to rectify the configuration  [here](https://github.com/commercetools/commercetools-jvm-sdk/blob/66a3ce79b10b97d6deef1a19a8a3586db015fac2/osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java)  but i opted for the easier way c7e02e7 as not to run them in the osgi container.

Reply:
not the best solution but its a fast one, and anyways what matters is that the integration tests run fine in OSGi which they still do.

Reply:
also this [one](https://github.com/commercetools/commercetools-jvm-sdk/pull/1958/files#diff-e5c9294211de558467e60b73075779faR45) is still `private` and not failing.

Is it not because of the `Mockito#mock` usage? (These three failing tests are using it)

Reply:
You are right, i just reviewed the logs and it seems so, its the Mockito dependency that shows in the error, i would expect `@NotOSGiCompatible` to fix the problem, since it prevents this test to be run in an OSGi container, if not i ll investigate the problem deeper

Reply:
> yep figured out i need to rectify the configuration here 

From the link you sent, seems like mockito dependency is there 🤔: https://github.com/commercetools/commercetools-jvm-sdk/blob/66a3ce79b10b97d6deef1a19a8a3586db015fac2/osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java#L75


> but i opted for the easier way c7e02e7 as not to run them in the osgi container.

hmm, but then it's weird that this test [`CategoryTest`](https://github.com/commercetools/commercetools-jvm-sdk/pull/1958/files#diff-e5c9294211de558467e60b73075779fa) runs successfully without this annotation..

> but i opted for the easier way c7e02e7 as not to run them in the osgi container.

shall we make the `private` again then?

Reply:
>hmm, but then it's weird that this test CategoryTest runs successfully without this annotation.

`CategoryTest`  actually doesn't have a dependency on Mockito as an import, it rather deserializes resources from json, this is why it runs fine

> From the link you sent, seems like mockito dependency is there

yep but in OSGi you have to import it in the `commercetools-sdk-base`'s manifest, which is not done since it just a test depedency, i m sure there is a solution for that, but i still opt for the easy `@NotOSGiCompatible` to unblock you faster.

Reply:
> CategoryTest actually doesn't have a dependency on Mockito as an import, it rather deserializes resources from json, this is why it runs fine

True, right :) 

> yep but in OSGi you have to import it in the commercetools-sdk-base's manifest, which is not done since it just a test depedency, i m sure there is a solution for that, but i still opt for the easy @NotOSGiCompatible to unblock you faster.

Ok, I am not so into this topic. But thanks for the explanation. 

> definitely, i ll revert my changes
👍